### PR TITLE
Cleanup: to prevent calling super on grandparents in PootleDetailView descendants

### PR DESCRIPTION
**Pull request by phlax**
- adds a required_permission var in `requires_permission` decorator
- updates the `ProjectsTranslateView` to use required_permission

### CODE REVIEW

File: pootle/apps/pootle_project/views.py [hunk 1]
```diff
@@ -374,11 +373,11 @@ def get(self, *args, **kwargs):
...
 
 class ProjectsTranslateView(ProjectsMixin, PootleTranslateView):
+    required_permission = "administrate"
 
-    @set_permissions
-    @requires_permission("administrate")
     def dispatch(self, request, *args, **kwargs):
-        return super(PootleDetailView, self).dispatch(request, *args, **kwargs)
+        return super(
+            ProjectsTranslateView, self).dispatch(request, *args, **kwargs)
```
Comment (unho) [inline pootle/apps/pootle_project/views.py]: This is unrelated change, please undo.

File: pootle/core/views.py [hunk 2]
```diff
@@ -92,6 +92,9 @@ def class_wrapper(f):
         def method_wrapper(self, request, *args, **kwargs):
             directory_permission = check_directory_permission(
                 permission, request, self.permission_context)
+            if directory_permission and self.view_permission:
```
Comment (unho) [inline pootle/core/views.py]: What is `self.view_permission`?

File: pootle/apps/pootle_project/views.py (hunk 1 above)
Comment (julen) [inline pootle/apps/pootle_project/views.py]: Funnily this seems to be the only change relevant to the PR title 😄

Comment (unho) [inline pootle/apps/pootle_project/views.py]: Right. Sorry for the noise. It seems I misread the code.

File: pootle/core/views.py [hunk 3]
```diff
@@ -92,6 +92,9 @@ def class_wrapper(f):
         def method_wrapper(self, request, *args, **kwargs):
             directory_permission = check_directory_permission(
                 permission, request, self.permission_context)
+            if directory_permission and self.required_permission:
```
Comment (unho) [inline pootle/core/views.py]: Why is it necessary to check for `directory_permission` if it is being overwritten?

Comment (phlax) [inline pootle/core/views.py]: we dont want to check the 2nd permission if the first has already failed

**Comment by unho** [comment]
lgtm

File: pootle/core/views.py [hunk 4]
```diff
@@ -92,6 +92,9 @@ def class_wrapper(f):
         def method_wrapper(self, request, *args, **kwargs):
             directory_permission = check_directory_permission(
                 permission, request, self.permission_context)
+            if directory_permission and hasattr(self, "required_permission"):
```
Comment (ta2-1) [inline pootle/core/views.py]: @phlax  `and self.required_permission != permission` ?

Comment (phlax) [inline pootle/core/views.py]: sounds like a good idea
